omnibus/verification/verify.rb

```diff
@@ -0,0 +1,548 @@
+
+      option :integration,
+        long: "--integration",
+        description: "Run integration tests. Possibly dangerous, for development systems only"
+
+      option :verbose,
+        long: "--verbose",
+        description: "Display all test output, not just failing tests"
+
+      class << self
+        def add_component(name, _delete_me = nil)
+          component = ComponentTest.new(name)
+          yield component if block_given? # delete this conditional
+          component_map[name] = component
+        end
+
+        def component(name)
+          component_map[name]
+        end
+
+        def components
+          component_map.values
+        end
+
+        def component_map
+          @component_map ||= {}
+        end
+      end
+
+      def components
+        self.class.components
+      end
+
+      bundle_install_mutex = Mutex.new
+
+      #
+      # Components included in Chef Development kit:
+      # :base_dir => Relative path of the component w.r.t. omnibus_apps_dir
+      # :gem_base_dir => Takes a gem name instead and uses first gem found
+      # :test_cmd => Test command to be launched for the component
+      #
+      add_component "berkshelf" do |c|
+        c.gem_base_dir = "berkshelf"
+        # For berks the real command to run is "#{embedded_bin("bundle")} exec thor spec:ci"
+        # We can't run it right now since graphviz specs are included in the
+        # test suite by default. We will be able to switch to that command when/if
+        # Graphviz is added to omnibus.
+        c.unit_test do
+          bundle_install_mutex.synchronize { sh("#{embedded_bin("bundle")} install") }
+          sh("#{embedded_bin("bundle")} exec #{embedded_bin("rspec")} --color --format progress spec/unit --tag ~graphviz")
+        end
+
+        # See older versions of this file in git to retrieve
+        # our cucumber test command for berkshelf.
+        # We had to remove them because we no longer bundle These tests cannot be run unless we bundle cucumber
+        # TODO mp 2019-09-09 disabled this until we get cucumber sorted.
+        # c.integration_test do
+        #   bundle_install_mutex.synchronize { sh("#{embedded_bin("bundle")} install") }
+        #   sh("#{embedded_bin("bundle")} exec #{embedded_bin("cucumber")} --color --format progress --tags ~@no_run --tags ~@spawn --tags ~@graphviz --strict")
+        # end
+
+        c.smoke_test do
+          tmpdir do |cwd|
+            FileUtils.touch(File.join(cwd, "Berksfile"))
+            sh("#{bin("berks")} install", cwd: cwd)
+          end
+        end
+      end
+
+      add_component "test-kitchen" do |c|
+        c.gem_base_dir = "test-kitchen"
+        c.unit_test do
+          bundle_install_mutex.synchronize { sh("#{embedded_bin("bundle")} install") }
+          sh("#{embedded_bin("bundle")} exec rake unit")
+        end
+        # TODO mp 2019-09-09 disabled this until we get cucumber sorted.
```

Comment:
Yep, I'm un-commenting these and will create a card to track the fact that we can only run smoke tests. Because we do only run smoke tests, uncommenting  will have no effect on verification.